src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,33 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (victim.hidden || victim instanceof Carnivore) {
```

Comment:
[FUNCTION NAMING] - The `hide` method should ensure that the `hidden` property is set to `true`, which is correctly implemented. However, the current `bite` method's condition `victim.hidden` should be more specific to check if `victim` is an instance of `Herbivore` and not hidden. The condition `victim.hidden || victim instanceof Carnivore` may lead to confusion as it mixes the check for a property that only exists on `Herbivore` with a type check for `Carnivore`. Consider restructuring the condition to first check if `victim` is an instance of `Herbivore` and then whether it is hidden.